### PR TITLE
RavenDB-19281 Add threshold to start flush and sync when performing replacement of side-by-side index.

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19281

### Additional description

There is no need to run it for small journals. 

### Type of change

- Optimization

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- This change requires a documentation update. 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Config/Categories/IndexingConfiguration.cs [hunk 1]
```diff
@@ -449,6 +449,14 @@ public virtual bool RunInMemory
         [IndexUpdateType(IndexUpdateType.None)]
         [ConfigurationEntry("Indexing.MaxTimeToWaitAfterFlushAndSyncWhenReplacingSideBySideIndexInSec", ConfigurationEntryScope.ServerWideOrPerDatabaseOrPerIndex)]
         public TimeSetting MaxTimeToWaitAfterFlushAndSyncWhenReplacingSideBySideIndex { get; protected set; }
+
+        [Description("Minimum total size of journals to run flush and sync when replacing side by side index.")]
+        [DefaultValue(1024)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/IndexingConfiguration.cs]: That is pretty big, no?

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 2]
```diff
@@ -1812,8 +1812,11 @@ public ReplaceStatus ReplaceIfNeeded(bool batchCompleted, bool didWork)
 
                     if (batchCompleted)
                     {
+                        var currentTotalSizeOfJournalsInBytes = _environment.Journal.Files.Select(i => 4 * 1024 * (i.Available4Kbs + i.WritePosIn4KbPosition)).Sum();
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/Index.cs]: better to expose in JournalFile the Size using the IJournalWriter.NumberOfAllocated4Kb IMO?
Also this should at the end be of type Size, so you can easily compare to the MinimumTotalSizeOfJournalsToRunFlushAndSyncWhenReplacingSideBySideIndex

Comment (maciejaszyk) [reply src/Raven.Server/Documents/Indexes/Index.cs]: Changed :)

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Server/Config/Categories/IndexingConfiguration.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Server/Config/Categories/IndexingConfiguration.cs]: It has been changed to 512 MB

**Pull request merged**